docs/source/format/Columnar.rst

```diff
@@ -0,0 +1,1266 @@
+
+      * Offsets buffer (int32)
+
+        | Bytes 0-27           | Bytes 28-63 |
+        |----------------------|-------------|
+        | 0, 2, 4, 7, 7, 8, 10 | unspecified |
+
+      * Values array (Int8):
+        * Length: 10, Null count: 0
+        * Null bitmap buffer: Not required
+
+          | Bytes 0-9                     | Bytes 10-63 |
+          |-------------------------------|-------------|
+          | 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 | unspecified |
+
+Note that while the inner offsets buffer encodes the start position in
+the inner values array, the outer offsets buffer encodes the start
+position of corresponding outer element in the inner offsets buffer.
+
+Fixed-Size List
+---------------
+
+Fixed-Size List is a nested type in which each array slot contains a
+fixed-size sequence of values all having the same type (heterogeneity
+can be achieved through unions, described later).
+
+A fixed size list type is specified like ``FixedSizeList<T>[N]``,
+where ``T`` is any type (primitive or nested) and ``N`` is a 32-bit
+signed integer representing the length of the lists.
+
+A fixed size list array is represented by a values array, which is a
+child array of type T. T may also be a nested type. The value in slot
+``j`` of a fixed size list array is stored in an ``N``-long slice of
+the values array, starting at an offset of ``j * N``.
+
+**Example Layout: ``FixedSizeList<byte>[4]`` Array**
+
+Here we illustrate ``FixedSizeList<byte>[4]``.
+
+For an array of length 4 with respective values: ::
+
+    [[192, 168, 0, 12], null, [192, 168, 0, 25], [192, 168, 0, 1]]
+
+will have the following representation: ::
+
+    * Length: 4, Null count: 1
+    * Null bitmap buffer:
+
+      | Byte 0 (validity bitmap) | Bytes 1-63            |
+      |--------------------------|-----------------------|
+      | 00001101                 | 0 (padding)           |
+
+    * Values array (byte array):
+      * Length: 16,  Null count: 0
+      * Null bitmap buffer: Not required
+
+        | Bytes 0-3       | Bytes 4-7   | Bytes 8-15                      |
+        |-----------------|-------------|---------------------------------|
+        | 192, 168, 0, 12 | unspecified | 192, 168, 0, 25, 192, 168, 0, 1 |
+
+
+Struct
+------
+
+A struct is a nested type parameterized by an ordered sequence of
+types (which can all be distinct), called its fields. Typically the
+fields have names, but the names and their types are part of the type
+metadata, not the physical memory layout.
+
+A struct array does not have any additional allocated physical storage
+for its values.  A struct array must still have an allocated null
+bitmap, if it has one or more null values.
+
+Physically, a struct type has one child array for each field. The
+child arrays are independent and need not be adjacent to each other in
+memory.
+
+For example, the struct (field names shown here as strings for illustration
+purposes)::
+
+    Struct <
+      name: String (= List<char>),
```

Comment:
I think it's a bit confusing to mention `String` and `List<char>` as synonyms since `String` is non-nested while `List` is nested.
Also `String` wasn't introduced yet, just `Binary`.